daemon/container.go

```diff
@@ -870,7 +870,9 @@ func (container *Container) Copy(resource string) (io.ReadCloser, error) {
 
 	// Check if this is actually in a volume
 	for _, mnt := range container.VolumeMounts() {
-		if len(mnt.MountToPath) > 0 && strings.HasPrefix(resource, mnt.MountToPath[1:]) {
+		// Check the relative path of the resource to the volume.
+		// If the relative path of the resource starts with "../", that means the resource is not within the volume since it's in a parent dir
+		if relPath, err := filepath.Rel(mnt.MountToPath, filepath.Join("/", resource)); err == nil && !strings.HasPrefix(relPath, "../") {
```

Comment:
Yes, this is an issue currently, but probably not a major one.

Was thinking of replacing this with something like `container.MountWithVolumes()`, which would be similar to `container.Mount()`, but also bind-mount the volumes into the host's container fs path (and unmount when finished)... not sure.